contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -1246,22 +1246,30 @@ def _attach_markers(collection):
                 current_track.markers.remove(marker)
 
                 # determine new item to attach the marker to
-                target_item = target_track.child_at_time(marker.marked_range.start_time)
-                if target_item is None or not hasattr(target_item, 'markers'):
+                try:
+                    target_item = target_track.child_at_time(
+                        marker.marked_range.start_time
+                    )
+
+                    if target_item is None or not hasattr(target_item, 'markers'):
+                        target_item = target_track
+
+                    # transform marked range into new item range
+                    marked_start_local = current_track.transformed_time(
+                        marker.marked_range.start_time, target_item
+                    )
+
+                    marker.marked_range = otio.opentime.TimeRange(
+                        start_time=marked_start_local,
+                        duration=marker.marked_range.duration
+                    )
+
+                except otio.exceptions.CannotComputeAvailableRangeError:
                     target_item = target_track
```

Comment:
Can you leave some comments here explaining when the exception is expected?

Reply:
Added